department-of-veterans-affairs/abd-vro
# Add Temporary Station of Jurisdiction endpoint to scv-bip-api and mock-bip-claims-api

## What was the problem?

Current svc-bip-api does not contain endpoint to set the `temp_station_of_jurisdiction`

Associated tickets or Slack threads:
- #2175 
- Blocked by / Waiting on merges of:
  - #2144 
  - #2165

## How does this fix it?[^1]

This adds the endpoint for setting the temporary station of jurisdiction to the BipApiService and the associated rabbitMQ configurations/declarations and listeners. 
This follows the class paradigm as diagramed in  #2165

## How to test this PR
- Run the unit tests via `./gradlew :svc-bip-api:check`
- Start the platform base docker containers to start RabbitMq and start the BIP Mock container (see [Docker Compose](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Docker-Compose))
  - run the integration tests via `./gradlew :svc-bip-api:integrationTest`
  - build and compose the `svc-bip-api` docker and run a test by:
     - create a new queue on the `bip-api` exchange called `response`
     - open to the rabbitmq console to the [getClaimDetailsQueue](http://localhost:15672/#/queues/%2F/getClaimDetailsQueue) 
     - publish a message with the body `1000` and add a message property called `reply_to` with the value `response`
     - open the response queue page and perform the get message action
     - verify it is a 200 and the response is the mocked response containing: `{"statusCode":200,"statusMessage":"OK","claim":{"claimId":"1000","phase":"Claim Received","endProductCode":"400"}}`
     - open to the rabbitmq console to the [putTempStationOfJurisdictionQueue](http://localhost:15672/#/queues/%2F/putTempStationOfJurisdictionQueue) 
     - publish a message with the body `{"claimId":1000,"tempStationOfJurisdiction":"398"` and add a message property called `reply_to` with the value `response`
     - open the response queue page and perform the get message action
     - verify it is a 200 with the response body `{"statusCode":200,"statusMessage":"OK"}`
     - open to the rabbitmq console to the [getClaimDetailsQueue](http://localhost:15672/#/queues/%2F/getClaimDetailsQueue) 
     - publish a message with the body `1000` and add a message property called `reply_to` with the value `response`
     - open the response queue page and perform the get message action
     - verify it is a 200 and the response is the mocked response containing: `{"statusCode":200,"statusMessage":"OK","claim":{"claimId":"1000","phase":"Claim Received","endProductCode":"400","tempStationOfJurisdiction":"398"}}`

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
